<?xml version="1.0" encoding="UTF-8"?>
<launch>

  <!-- usage:  -->
  <!-- For mapping: roslaunch agimus_demos tiago_gerard_bauzil_tiago_gazebo.launch public_sim:=true navigation:=false
       For localization & navigation: 
          - copy the map data in gerard_bauzil packages to default ros map location
          - roslaunch agimus_demos tiago_gerard_bauzil_tiago_gazebo.launch public_sim:=true navigation:=true

  -->
  <arg name="namespace" default="/"/>
  <arg name="public_sim" default="true"/>
  <arg name="navigation" default="false"/>
  <arg name="robot" default="titanium"/>
  <arg name="arm" default="$(eval {'iron': False}.get(arg('robot'), True))"/>
  <arg name="end_effector" default="$(eval {'iron': 'false', 'steel': 'pal-gripper', 'titanium': 'pal-hey5'}.get(arg('robot'), 'pal-gripper'))"/>
  <arg name="ft_sensor" default="$(eval {'titanium': 'schunk-ft'}.get(arg('robot'), 'false'))"/>
  <arg name="laser_model" default="sick-571"/>
  <arg name="camera_model" default="orbbec-astra"/>

  <arg name="world" default="empty"/>
  <arg name="gzpose" default="-x 0.0  -y   0.0 -z  0.0   -R 0.0 -P 0.0 -Y  0.0" unless="$(arg navigation)"/>
  <arg name="gzpose" default="-x 1.78 -y 2.1 -z -0.003 -R 0.0 -P 0.0 -Y 70.0" if="$(arg navigation)"/>
  <arg name="tuck_arm" default="true"/>

  <arg name="planner" default="base"/>
  <arg name="global_planner" default="global_planner"/>
  <arg name="local_planner" default="$(eval 'teb' if arg('public_sim') else 'pal')"/>
  <arg name="mapping" default="gmapping" />
  <arg name="map" default="$(env HOME)/.pal/tiago_maps/config"/>
  <arg name="localization" default="amcl"/>

  <arg name="rviz" default="true"/>
  <arg name="gzclient" default="true"/>
  <arg name="recording" default="false"/>
  <arg name="extra_gazebo_args" default=""/>

  <arg name="rgbd_sensors" default="false"/>

  <arg name="sim_sufix" value="_public_sim" if="$(arg public_sim)"/>
  <arg name="sim_sufix" value="" unless="$(arg public_sim)"/>

  <env name="PAL_HOST" value="tiago" />


  <group ns="$(arg namespace)">

    <include file="$(find tiago_gazebo)/launch/tiago_gazebo.launch">
      <arg name="world" value="$(arg world)"/>
      <arg name="arm" value="$(arg arm)"/>
      <arg name="end_effector" value="$(arg end_effector)"/>
      <arg name="ft_sensor" value="$(arg ft_sensor)"/>
      <arg name="laser_model" value="$(arg laser_model)"/>
      <arg name="camera_model" value="$(arg camera_model)"/>
      <arg name="gzpose" value="$(arg gzpose)"/>
      <arg name="tuck_arm" value="$(arg tuck_arm)"/>
      <arg name="gui" value="$(arg gzclient)"/>
      <arg name="public_sim" value="$(arg public_sim)"/>
      <arg name="recording" value="$(arg recording)"/>
      <arg name="extra_gazebo_args" value="$(arg extra_gazebo_args)"/>
    </include>


    <node name="rviz" pkg="rviz" type="rviz" if="$(arg rviz)" args="-d $(find tiago_2dnav)/config/rviz/navigation$(arg sim_sufix).rviz"/>
  </group>


 <!-- salle Gerard Bauzil -->
  <group ns="gerard_bauzil">
    <arg name="room_pose" default="-X 0.0 -Y 0.0 -Z -0.0 -R 0.0 -P 0.0 -Y 0"/>
    <param name="robot_description" textfile="$(find gerard_bauzil)/urdf/gerard_bauzil.urdf"/>
    <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher" />
    <node name="spawn_room" pkg="gazebo_ros" type="spawn_model" args="-file $(find gerard_bauzil)/urdf/gerard_bauzil.urdf -urdf -model gerard_bauzil -X -0 -Y 0. -Z -0.0 -R 0.0 -P 0.0 -Y 0" />
  </group>

  
  <!-- pointcloud to laserscan-->
  <group if="$(arg public_sim)">
    <include file="$(find tiago_laser_sensors)/launch/rgbd_cloud_laser.launch">
      <arg name="cloud" value="/xtion/depth_registered/points"/>
    </include>
  </group>


  <include file="$(find tiago_2dnav_gazebo)/launch/navigation.launch">
    
    <!--for localisation & navigation-->
    <arg name="state" value="localization" if="$(arg navigation)"/>
    
     <!--for mapping-->
    <arg name="state" value="mapping" unless="$(arg navigation)"/>
    
    <arg name="planner" value="$(arg planner)"/>
    <arg name="global_planner" value="$(arg global_planner)"/>
    <arg name="local_planner" value="$(arg local_planner)"/>
    <!-- <arg name="localization" value="$(arg navigation)"/> -->
    <arg name="map" value="$(arg map)"/>

    <!-- only use gmapping -->
    <arg name="mapping" value="gmapping"/> 
    
    <arg name="public_sim" value="$(arg public_sim)"/>
    <arg name="rgbd_sensors" value="$(arg rgbd_sensors)"/>
    <arg name="laser_model" value="$(arg laser_model)"/>
  </include>

  <group unless="$(arg public_sim)">
    <include file="$(find pal_loc_measure)/launch/pal_loc_measure.launch">
      <arg name="training" value="False"/>
    </include>
  </group>

 
</launch>

                    